src/funding/funding.ts

```diff
@@ -0,0 +1,428 @@
+        this.depositAmount = remainingNeeded;
+      }
+    }
+  }
+
+  /**
+   * Deposits the tokens from the wallet to the contract
+   */
+  private depositTokens(): void {
+    const tokenSymbol = this.deal.daoRelatedToWallet.tokens[this.selectedToken].symbol;
+    //TODO re-check the contract to validate how many tokens are needed for the required deposit amount
+    const recentRequiredTokens = converter.fromView(10, 18);
+    //TODO re-check the balance of the wallet to make sure the wallet has enough tokens
+    const recentWalletBalance = converter.fromView(120, 18);
+    //rebind token data if it's changed
+    //TODO reset all the data after checking
+    // const token = this.deal.daoRelatedToWallet.tokens[this.selectedToken] as ITokenFunding;
+    // token.required = recentRequiredTokens;
+    // token.deposited = converter.fromView(120);
+    // token.target = converter.fromView(120);
+    // token.percentCompleted = 12;
+
+    if (Number(this.depositAmount) > Number(recentWalletBalance)) {
+      this.eventAggregator.publish("handleInfo", new EventConfig(`The amount you wish to deposit (${converter.toView(this.depositAmount, 18)} ${tokenSymbol}) exceeds the current balance in your wallet (${converter.toView(recentWalletBalance, 18)} ${tokenSymbol}). Please submit again.`, EventMessageType.Warning, "Insufficient Balance"));
+      this.depositAmount = recentWalletBalance;
+      return;
+    }
+    if (Number(this.depositAmount) > Number(recentRequiredTokens)) {
+      this.eventAggregator.publish("handleInfo", new EventConfig(`The amount you wish to deposit (${converter.toView(this.depositAmount, 18)} ${tokenSymbol}) exceeds the required funding needed (${converter.toView(recentRequiredTokens, 18)} ${tokenSymbol}). Please submit again.`, EventMessageType.Warning));
+      this.depositAmount = recentRequiredTokens;
+      return;
+    }
+    //TODO implement the deposit of tokens
+    this.eventAggregator.publish("handleInfo", new EventConfig(`Depositing ${converter.toView(this.depositAmount, 18)} ${tokenSymbol} on behalf of ${this.deal.daoRelatedToWallet.name}`, EventMessageType.Info, "Deposit Submitted"));
+    //TODO handle wallet provider transaction rejection
+    //TODO handle the popup notification on the event of the deposit actually being completed
+  }
+
+  /**
+   * Gets the HTML for the dao token swap modal popup
+   * @param dao
+   * @returns string
+   */
+  private getDaoHtmlForSwap(dao: IDAO): string {
+    return `
+      <div>
+        <h6>${dao.name} treasury address</h6>
+        <div class="dao">
+          ${dao.treasury_address} 
+          <div class="buttons">
+            <copy-to-clipboard-button text-to-copy="${dao.treasury_address}"></copy-to-clipboard-button>
+            <svg class="etherscan-button" ptooltip="Inspect on Etherscan" click.delegate="data.gotoEtherscan('${dao.treasury_address}')" width="17" height="17" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none">
```

Comment:
That's the svg for the arrow icon when clicking on an external link. This code will get moved to the deal dashboard page because swapping is moving there and I can make the icon into a component.

Reply:
I will clean this up to use the etherscan button

Reply:
This has been updated to use the <etherscan-button> component